### PR TITLE
add more logs

### PR DESCRIPTION
**Pull request by weekface**
### Check List 

Tests 

 - Unit test
 - No code

Code changes

 - Has Go code change

Side effects

Related changes

 - Need to cherry-pick to the release branch

### Does this PR introduce a user-facing change?:
 
 ```release-note
NONE
 ```

### CODE REVIEW

File: pkg/manager/member/tidb_failover.go [hunk 1]
```diff
@@ -40,6 +41,7 @@ func (tf *tidbFailover) Failover(tc *v1alpha1.TidbCluster) error {
 		_, exist := tc.Status.TiDB.FailureMembers[tidbMember.Name]
 		if exist && tidbMember.Health {
 			delete(tc.Status.TiDB.FailureMembers, tidbMember.Name)
+			glog.Errorf("tidb failover: delete %s from tidb failoverMembers", tidbMember.Name)
```
Comment (tennix) [inline pkg/manager/member/tidb_failover.go]: Is this an error log?

**Comment by weekface** [comment]
all these logs are "state changes", they can help us diagnose problems

File: pkg/manager/member/utils.go [hunk 2]
```diff
@@ -182,6 +182,7 @@ func serviceEqual(new, old *corev1.Service) (bool, error) {
 // setUpgradePartition set statefulSet's rolling update partition
 func setUpgradePartition(set *apps.StatefulSet, upgradeOrdinal int32) {
 	set.Spec.UpdateStrategy.RollingUpdate = &apps.RollingUpdateStatefulSetStrategy{Partition: &upgradeOrdinal}
+	glog.Infof("set %s/%s partition to %d successfully", set.GetNamespace(), set.GetName(), upgradeOrdinal)
```
Comment (cofyc) [inline pkg/manager/member/utils.go]: I think it's not a good idea to log in the utility function. log when the API operation has done?

File: pkg/manager/member/orphan_pods_cleaner.go [hunk 3]
```diff
@@ -90,8 +91,10 @@ func (opc *orphanPodsCleaner) Clean(tc *v1alpha1.TidbCluster) (map[string]string
 
 		err = opc.podControl.DeletePod(tc, pod)
 		if err != nil {
+			glog.Errorf("orphan pods cleaner: failed to clean orphan pod: %s/%s, %v", ns, podName, err)
```
Comment (gregwebs) [inline pkg/manager/member/orphan_pods_cleaner.go]: I generally prefer to annotate errors: that way there is a single error data structure with all the information that can be logged in one spot, and if there is an API request sent back in the response.

Comment (weekface) [reply pkg/manager/member/orphan_pods_cleaner.go]: Yes, our error logs style need a lot of improvements. We may discuss it in a new issue?

This pr tries to add detailed logs when something changes to help us diagnose problems.

Comment (gregwebs) [reply pkg/manager/member/orphan_pods_cleaner.go]: It is up to you. I think either way you should get a detailed log to help diagnose problems. If there is concurrent logging annotation will be simpler to understand.

File: pkg/scheduler/predicates/ha.go [hunk 4]
```diff
@@ -128,18 +129,21 @@ func (h *ha) Filter(instanceName string, pod *apiv1.Pod, nodes []apiv1.Node) ([]
 		// replicas less than 3 cannot achieve high availability
 		if replicas < 3 {
 			minNodeNames = append(minNodeNames, nodeName)
+			glog.Infof("replicas is %d, add node %s to minNodeNames", replicas, nodeName)
```
Comment (cofyc) [inline pkg/scheduler/predicates/ha.go]: Logs print here are unnecessary, at line 109 we know the replicas of the component, if the replicas is smaller than 3, all nodes will be added into `minNodeNames`. 

How about making it simpler, we can skip the for loop if the replicas is smaller than 3 and print an informative log?

File: pkg/scheduler/predicates/ha.go [hunk 5]
```diff
@@ -128,18 +129,21 @@ func (h *ha) Filter(instanceName string, pod *apiv1.Pod, nodes []apiv1.Node) ([]
...
 			continue
 		}
 
 		podsCount := len(podNames)
 		if podsCount+1 >= int(replicas+1)/2 {
+			glog.Infof("node %s podsCount is %d, skipping", nodeName, podsCount)
```
Comment (cofyc) [inline pkg/scheduler/predicates/ha.go]: print the `int(replicas+1)/2` too?

File: pkg/manager/member/orphan_pods_cleaner.go (hunk 3 above)
Comment (weekface) [reply pkg/manager/member/orphan_pods_cleaner.go]: what is "logging annotation"?

File: pkg/scheduler/predicates/ha.go (hunk 4 above)
Comment (weekface) [reply pkg/scheduler/predicates/ha.go]: How to skip the `for` loop? we must range the `nodeMap` anyway

File: pkg/scheduler/predicates/ha.go (hunk 5 above)
Comment (weekface) [reply pkg/scheduler/predicates/ha.go]: `replicas` has been printed in 109.

File: pkg/manager/member/utils.go (hunk 2 above)
Comment (weekface) [reply pkg/manager/member/utils.go]: removed the `successfully ` word. ptal

File: pkg/scheduler/predicates/ha.go (hunk 5 above)
Comment (weekface) [reply pkg/scheduler/predicates/ha.go]: addressed.

File: pkg/scheduler/predicates/ha.go (hunk 4 above)
Comment (cofyc) [reply pkg/scheduler/predicates/ha.go]: hmm, you're right, we need to get node name

**Pull request merged**